janusgraph-test/src/test/java/org/janusgraph/diskstorage/keycolumnvalue/inmemory/BufferPageTest.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.keycolumnvalue.inmemory;
+
+import org.janusgraph.diskstorage.Entry;
+import org.janusgraph.diskstorage.keycolumnvalue.inmemory.BufferPage;
+import org.janusgraph.diskstorage.util.StaticArrayBuffer;
+import org.janusgraph.diskstorage.util.StaticArrayEntry;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.List;
+
+public class BufferPageTest
+{
+    private static final String COL_START = "00colStart";
+    private static final String COL_END = "99colEnd";
+    private static final String VERY_END = "zz";
+
+    @Test
+    public void testGetIndex()
+    {
+        Entry[] entries = new Entry[]
+                {
+                    StaticArrayEntry.of(StaticArrayBuffer.of(COL_START.getBytes()), StaticArrayBuffer.of("valStart".getBytes())),
```

Comment:
You can simplify the code if you make a separate method like
```
private Entry makeEntry(String something1, String something2){
    return StaticArrayEntry.of(StaticArrayBuffer.of(something1.getBytes()), StaticArrayBuffer.of(something2.getBytes()))
}
```
Then you can use:
```
new Entry[]{
  makeEntry(COL_START, "valStart"),
  makeEntry("01col1", "val1"),
  ...
}
```

Reply:
@porunov thanks for your comments. I went through them and they look to be mostly "cosmetic" so far. I am happy to discuss them further, and regardless of the results of the discussion or my personal opinion, ultimately I am happy to make any change of this type if that is going to be the condition for approval. 

However I suggest that we return to these when/if the actual logic and performance are in fact reviewed and accepted in general.

Reply:
@dk-github Sure. I agree here with you

Reply:
same opinion with @porunov , will make this class easier to read

Reply:
done